### PR TITLE
Improved the SVG graphic and code.

### PR DESCRIPTION
**Pull request by denilsonsa**

### CODE REVIEW

File: coderwall.svg [hunk 1]
```diff
@@ -1,58 +1,10 @@
...
-     style="fill:#000000;fill-opacity:1;stroke:#000000;stroke-opacity:1" />
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Drawn from scratch, based on coderwall.png from https://assembly.com/coderwall/assets -->
+<svg version="1.1" viewBox="0 0 300 300" xmlns="http://www.w3.org/2000/svg">
+<circle cx="30" cy="30" r="30" fill="#3e8dcc"/>
+<circle cx="150" cy="30" r="30" fill="#3e8dcc"/>
+<circle cx="270" cy="30" r="30" fill="#3e8dcc"/>
+<circle cx="150" cy="150" r="30" fill="#3e8dcc"/>
+<circle cx="270" cy="150" r="30" fill="#3e8dcc"/>
+<circle cx="270" cy="270" r="30" fill="#3e8dcc"/>
```
Comment (twolfson) [inline coderwall.svg]: This doesn't seem to be rendering in GitHub's SVG preview. Can you find out what's wrong with that?

File: coderwall.svg [hunk 2]
```diff
@@ -1,58 +1,10 @@
...
-     style="fill:#000000;fill-opacity:1;stroke:#000000;stroke-opacity:1" />
-  <path
-     d="m 10.247191,7.9550562 a 3.9101124,3.775281 0 1 1 -7.8202248,0 3.9101124,3.775281 0 1 1 7.8202248,0 z"
-     transform="matrix(0.67889236,0,0,0.70313852,17.897806,15.606494)"
-     id="path3049-35-6"
-     style="fill:#000000;fill-opacity:1;stroke:#000000;stroke-opacity:1" />
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Drawn from scratch, based on coderwall.png from https://assembly.com/coderwall/assets -->
+<svg version="1.1" viewBox="0 0 300 300" xmlns="http://www.w3.org/2000/svg">
+<circle cx="30" cy="30" r="30" fill="#3e8dcc"/>
```
Comment (twolfson) [inline coderwall.svg]: Please indent these `circle` tags. I prefer XML tags to have indentation for each level of depth.

File: README.md [hunk 3]
```diff
@@ -1,10 +1,5 @@
-Coderwall SVG [![Donate on Gittip](http://badgr.co/gittip/twolfson.png)](https://www.gittip.com/twolfson/)
+Coderwall SVG [![Donate on Gittip](https://img.shields.io/gratipay/twolfson.svg)](https://gratipay.com/~twolfson/)
```
Comment (twolfson) [inline README.md]: This pattern is very out of date with how I typically link to gratipay. Please leave this in its original form and I will update it accordingly in another PR/release

File: README.md [hunk 4]
```diff
@@ -1,10 +1,5 @@
...
 =============
-In this repo is a quick (and sloppy) conversion of the [Coderwall](http://coderwall.com/) logo to SVG
+In this repo is a quick conversion of the [Coderwall](http://coderwall.com/) logo to SVG
 
-![Coderwall logo](https://raw.github.com/twolfson/coderwall-svg/master/coderwall.png)
-
-Download Links
---------------
-
-[svg](https://raw.github.com/twolfson/coderwall-svg/master/coderwall.svg) [png](https://raw.github.com/twolfson/coderwall-svg/master/coderwall.png)
```
Comment (twolfson) [inline README.md:10]: Please move back to the original split format of logo and download links. It was much easier for someone to get a link to download the resource (e.g. don't need to click to the page then click "raw" then click download).

File: README.md [hunk 5]
```diff
@@ -1,10 +1,5 @@
...
\ No newline at end of file
+[![Coderwall logo](https://cdn.rawgit.com/denilsonsa/coderwall-svg/88921c6211e60c960cd8c154068e1688affaa97b/coderwall.svg)](coderwall.svg)
```
Comment (twolfson) [inline README.md]: Please move back to `raw.github.com`, this was intentional as:
- There is no caching when viewing the resource meaning `master` is always up to date
- We don't need to worry about bumping the SHA on every edit to the SVG

**Comment by twolfson** [comment]
This PR was very aggressive and prob could have avoided a lot of work if an issue was opened first. The majority of requests are to move everything back to their original format and only update the SVG.

File: coderwall.svg (hunk 1 above)
Comment (denilsonsa) [inline coderwall.svg]: [Works fine for me now](https://github.com/denilsonsa/coderwall-svg/blob/master/coderwall.svg). Earlier today GitHub had some server issues.

**Comment by denilsonsa** [comment]
> This PR was very aggressive and prob could have avoided a lot of work if an issue was opened first. The majority of requests are to move everything back to their original format and only update the SVG.

Don't worry. Reverting the `README.md` file is easy. And the SVG file has been updated.

File: coderwall.svg [hunk 6]
```diff
@@ -1,58 +1,9 @@
...
-     id="path3049-35-6"
-     style="fill:#000000;fill-opacity:1;stroke:#000000;stroke-opacity:1" />
+<?xml version="1.0" encoding="utf-8"?>
+<svg version="1.1" viewBox="0 0 300 300" xmlns="http://www.w3.org/2000/svg">
+	<circle cx="30" cy="30" r="30" fill="#3e8dcc"/>
+	<circle cx="150" cy="30" r="30" fill="#3e8dcc"/>
+	<circle cx="270" cy="30" r="30" fill="#3e8dcc"/>
+	<circle cx="150" cy="150" r="30" fill="#3e8dcc"/>
+	<circle cx="270" cy="150" r="30" fill="#3e8dcc"/>
+	<circle cx="270" cy="270" r="30" fill="#3e8dcc"/>
```
Comment (twolfson) [inline coderwall.svg]: Please use spaces for indentation as that was the existing coding style for this repo

File: coderwall.svg (hunk 1 above)
Comment (twolfson) [inline coderwall.svg]: It's strange. It wasn't loading in Firefox on OS X but is loading in Firefox on Linux fine. Weird.

**Comment by denilsonsa** [comment]
Done.

**Comment by twolfson** [comment]
Sorry to keep on making this a back and forth. It looks like there is no whitespace padding as part of the SVG but in most scenarios, the logo has padding. For example:

https://assembly.firesize.com/120x120/g_center/https://d1015h9unskp4y.cloudfront.net/attachments/1e4aebe6-4771-45da-8805-693787bfe644/coderwall.png

Is there anything in the Coderwall style guide about how much padding is required?

**Comment by denilsonsa** [comment]
There is no actual official style guide. I've rewritten the dimensions to the closest-to-official I could get.

File: coderwall.svg [hunk 7]
```diff
@@ -1,58 +1,9 @@
...
-     id="path3049-35-7"
-     style="fill:#000000;fill-opacity:1;stroke:#000000;stroke-opacity:1" />
-  <path
-     d="m 10.247191,7.9550562 a 3.9101124,3.775281 0 1 1 -7.8202248,0 3.9101124,3.775281 0 1 1 7.8202248,0 z"
-     transform="matrix(0.67889236,0,0,0.70313852,17.897806,15.606494)"
-     id="path3049-35-6"
-     style="fill:#000000;fill-opacity:1;stroke:#000000;stroke-opacity:1" />
+<?xml version="1.0" encoding="utf-8"?>
+<svg version="1.1" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
+  <circle cx="5" cy="5" r="1.4142" fill="#3e8dcc"/>
```
Comment (twolfson) [inline coderwall.svg:3]: For our reference, what fraction is `1.4142`?

**Comment by twolfson** [comment]
Looks great, thanks for working on this =)

File: coderwall.svg (hunk 7 above)
Comment (denilsonsa) [inline coderwall.svg:3]: Analyzing the CSS from http://codepen.io/TimPietrusky/pen/hLolB, the entire bg has `2em` of width, and each dot is `1em`, with `0.5em` offset from each one. Multiplying all by 10, we have the `viewBox="0 0 20 20"` and each center on `5` SVG points increments.

The radius was a bit more tricky. The CSS used `20%`; but what does that percentage mean? It is measured from the center to the furthest corner, which is the diagonal of the square. `20% × 0.5 × √2 ≅ 0.14142135623`

I'm pretty sure that `20%` is just some approximation by the author of that CSS example, but all "official" logos are low-quality raster pictures anyway.

Comment (twolfson) [inline coderwall.svg:3]: Cool, thanks for the info =)

**Pull request merged**